registry/cache/warming.go

```diff
@@ -284,6 +288,12 @@ func (w *Warmer) warm(ctx context.Context, now time.Time, logger log.Logger, id
 			go func(update update) {
 				defer func() { awaitFetchers.Done(); <-fetchers }()
 
+				select {
+				case <-ctxc.Done():
+					return // terminate on error
+				default: // avoid blocking
```

Comment:
Having expanded the unchanged lines above, I don't think you need this here, since

 1. it'll exit the loop entirely when the context has been cancelled, thanks to L282-286
 2. any goroutines still underway when the context is cancelled will either complete or immediately return an error from `client.Manifest()`

This early exit will only ever be hit if the goroutine was created before the context was cancelled, and run after it was cancelled. Which will be the case for fewer (probably many fewer) than w.Burst goroutines. I'd rather have less code than a very slightly earlier exit.